ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -565,24 +565,28 @@ public int compare(Task t1, Task t2) {
   };
 
   private class SortTableHeaderRenderer implements TableCellRenderer {
-    private Icon ascIcon;
-    private Icon descIcon;
-    private TableCellRenderer defaultRenderer;
+    private final Icon myAscIcon;
+    private final Icon myDescIcon;
+    private final TableCellRenderer myDefaultRenderer;
     SortTableHeaderRenderer(JTable t) {
-      defaultRenderer = t.getTableHeader().getDefaultRenderer();
-      ascIcon = UIManager.getIcon("Table.ascendingSortIcon");
-      descIcon = UIManager.getIcon("Table.descendingSortIcon");
+      myDefaultRenderer = t.getTableHeader().getDefaultRenderer();
+      myAscIcon = UIManager.getIcon("Table.ascendingSortIcon");
+      myDescIcon = UIManager.getIcon("Table.descendingSortIcon");
     }
 
     @Override
     public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int col) {
-      Component c = defaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, col);
+      Component c = myDefaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, col);
       ColumnImpl column = myTableHeaderFacade.findColumnByViewIndex(col);
       if (column.getSort() == SortOrder.ASCENDING) {
-        ((JLabel) c).setIcon(ascIcon);
+        if (c instanceof JLabel) {
```

Comment:
Please move this check to upper level, before you start dealing with columns and sort orders.